src/glob/GlobWalker.zig

```diff
@@ -1641,51 +1580,42 @@ pub fn GlobWalker_(
             arena: *Arena,
             patternComponents: *ArrayList(Component),
             pattern: []const u8,
-            out_cp_len: *u32,
-            out_pattern_cp: *[]u32,
             has_relative_patterns: *bool,
             end_byte_of_basename_excluding_special_syntax: *u32,
             basename_excluding_special_syntax_component_idx: *u32,
         ) !void {
-            var start_cp: u32 = 0;
             var start_byte: u32 = 0;
 
-            const iter = CodepointIterator.init(pattern);
-            var cursor = CodepointIterator.Cursor{};
-
-            var cp_len: u32 = 0;
             var prevIsBackslash = false;
             var saw_special = false;
-            while (iter.next(&cursor)) : (cp_len += 1) {
-                const c = cursor.c;
+            var i: u32 = 0;
+            var width: u32 = 0;
+            while (i < pattern.len) : (i += 1) {
+                const c = pattern[i];
+                width = bun.strings.utf8ByteSequenceLength(c);
```

Comment:
This should use the CodepointIterator because this code will not handle invalid surrogate pairs correctly on Windows